evm/src/executor/inspector/cheatcodes/mod.rs

```diff
@@ -445,6 +447,11 @@ where
                                 value: Some(call.transfer.value),
                                 data: Some(call.input.clone().into()),
                                 nonce: Some(account.info.nonce.into()),
+                                gas: if is_fixed_gas_limit {
+                                    Some(call.gas_limit.into())
+                                } else {
+                                    None
+                                },
```

Comment:
Note that we are setting the `gas` field on the `TransactionRequest` struct at the time of its construction here.
So we are not "overriding" anything on the call object. 

I can see if there is a way to override the field on the `call` object instead (as soon as possible to its construction), but I feel it might have a higher chance of unintended side effects elsewhere.

If this is purely about code aesthetics I could also replace the `check_if_fixed_gas_limit` helper function with `get_gas_limit_if_fixed` and change this here to:
`gas: get_gas_limit_if_fixed(data, call.gas_limit)`.